WP3/D3.8/report.tex

```diff
@@ -185,13 +185,94 @@ \section{Continuous integration achievements of OpenDreamKit projects}\label{pro
 
 \subsection{SageMath}
 
-% Julian should remark on his efforts here: In particular what problems
-% motivated his efforts, and how he thinks this will improve things in the
-% future.  Maybe make note here about producing Docker images as build
-% artifacts (TODO: confirm if GAP is doing this too), and further how
-% mybinder.org turns out to be a useful development tool in conjunction
-% with Dockerized builds.  Not sure how much to say about that here, or
-% leave it to the conclusion.
+The SageMath project had traditionally employed two distinct CI systems, the
+Buildbot and Sage's own development, the
+patchbot\footnote{\url{https://github.com/sagemath/sage-patchbot}}.
+
+\subsubsection{Buildbot and patchbot}
+Sage's Buildbot runs as a non-advisory CI system on a wide variety of
+platforms. It builds Sage and runs its test suite for every change that gets
+accepted into the mainline of the Sage code. Its main purpose is not to be an
+advisory tool for developers but to catch problems and incompatibilities of
+Sage or its dependencies, in particular on more obscure platforms that
+dependent projects do often not test for. Due to the vast range of platforms
+covered, the Buildbot can not realistically be turned into an advisory CI
+system as it is not conceivable to provision and administer the number of
+machines that would be necessary to run Sage's test suite at every stage of a
+test that gets proposed by a developer.
+
+Therefore, the Sage community at some point developed its own patchbot system
+which runs Sage's test suite at every stage of a proposed change. Anybody can
+run a patchbot on their private computer, typically a Linux system. These
+patchbots will then look for proposed changes to Sage, run Sage's test suite,
+and report findings back to a central server. It is worth noting that the
+patchbot system predates most of the currently popular CI systems.
+
+The patchbot system comes with a number of issues: It is not nearly as polished
+as the professional offerings which means that its workings can often be
```

Comment:
I have a problem with the use of "professional offerings" in this case.  While it's certainly true that the patchbot is unpolished, I wouldn't consider it "unprofessional" either which this sort of implies by contrast.

I'm not sure what to suggest instead though.  Maybe "commercial offerings" would be valid, though it should be clear that "commercial" does not mean we have to pay for them (as all these services are free for open source projects--perhaps something I should clarify earlier in the report).

Reply:
You're right the wording is unfortunate. Maybe dedicated? The idea is that it's not homegrown...commercial is also valid I think.